### PR TITLE
Faceted browser fixes

### PR DESCRIPTION
**Pull request by jimmccusker**

### CODE REVIEW

File: filters.py [hunk 1]
```diff
@@ -370,29 +373,34 @@ def facet_values(facets, variables, constraints):
...
             constraints = json.loads(constraints)
+        else:
+            constraints = []
         if len(variables) > 0:
             variables = json.loads(variables)
+        else:
+            variables = []
         query = instance_data_template.render(constraints=constraints, variables=variables, this=this)
-        
+        print (query)
```
Comment (gordom6) [inline filters.py]: Smite.

File: static/js/whyis.js [hunk 2]
```diff
@@ -441,41 +441,63 @@ function whyis() {
         SmartFacetConstructor.prototype.setSelectedValue = setSelectedValue;
         SmartFacetConstructor.prototype.getConstraint = getConstraint;
 
+        SmartFacetConstructor.prototype.getState = function() {
```
Comment (gordom6) [inline static/js/whyis.js]: I'm pretty sure you could use an ES6 class here. Worth trying.

File: static/js/whyis.js [hunk 3]
```diff
@@ -441,41 +441,63 @@ function whyis() {
...
         function SmartFacetConstructor(options) {
-            
             BasicFacet.call(this, options);
             if (!this.config.multiType) this.config.multiType = "intersection";
+            this.selected = [];
+            this.keys = [$mdConstant.KEY_CODE.COMMA];
+            this.constraintTypes = ['Include','Exclude','Require','Show'];
         }
 
         function getSelectedValue() {
```
Comment (gordom6) [inline static/js/whyis.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get

File: static/js/whyis.js [hunk 4]
```diff
@@ -441,41 +441,63 @@ function whyis() {
...
-                    return values
-                        .map(function(v) { return ' ?id ' + self.predicate + ' ' + v + ' . ' } )
-                        .join("\n");
-                } else {
-                    var selectionConstraint = '?var_'+makeID();
-                    return ' ?id ' + self.predicate + ' '+ selectionConstraint + ' \n values ' +
-                        selectionConstraint + ' { '+ values.join(" ") + ' }';
+                var selectionConstraint = '?var_'+makeID();
+                var result = ' ?id ' + self.predicate + ' '+ selectionConstraint + '. \n';
+                var filteredValues = values.filter(function(d) { return d.value; }).map(function(d) {return d.value});
```
Comment (gordom6) [inline static/js/whyis.js]: ES6 arrow functions

File: static/js/whyis.js (hunk 2 above)
Comment (jimmccusker) [reply static/js/whyis.js]: This is building off of an existing framework that uses this approach. I need to remain compatible with it. We are partway through refactoring it away, so after that, we can take a look.

**Pull request merged**